### PR TITLE
Feat support external traffic policy

### PR DESCRIPTION
**Pull request by AhmedGrati**
# Description
This pull request solves #1543.
Since the `externalTrafficPolicy` is only specified when we have a `Service` of type `LoadBalancer` or `NodePort`, we will output a warning message when the user specifies another type of `Service`.
This feature is implemented for both: Kubernetes and Openshift.

### CODE REVIEW

**Comment by ZakariaNaaija** [comment]
Hi @AhmedGrati,
Thanks for handling the issue I have mentioned.

> Since the externalTrafficPolicy is only specified when we have a Service of type LoadBalancer, we will output a warning message when the user specifies another type of Service.

externalTrafficPolicy can be used along with a service of type NodePort.  
> Local preserves the client source IP and avoids a second hop for LoadBalancer and NodePort type Services, but risks potentially imbalanced traffic spreading.

[Source](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip)

**Comment by AhmedGrati** [comment]
> Hi @AhmedGrati, Thanks for handling the issue I have mentioned.
> 
> > Since the externalTrafficPolicy is only specified when we have a Service of type LoadBalancer, we will output a warning message when the user specifies another type of Service.
> 
> externalTrafficPolicy can be used along with a service of type NodePort.
> 
> > Local preserves the client source IP and avoids a second hop for LoadBalancer and NodePort type Services, but risks potentially imbalanced traffic spreading.
> 
> [Source](https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip)

Thank you for raising it, much appreciated. I will update it.

**Comment by mrsangjunboon2018** [comment]
FUCK!

**Comment by cdrage** [comment]
This will need a huge rebase, sorry for the work @AhmedGrati !

**Comment by AhmedGrati** [comment]
@cdrage It's ready to be reviewed.

**Review by cdrage — COMMENTED** [review]
The code LGTM but can you add some documentation as well for the user guide / under labels as this is a new label?

**Review by cdrage — COMMENTED** [review]
Just sayin', these tests are GREAT and thank you so much for always covering it all 💯 💯 💯

**Comment by AhmedGrati** [comment]
> The code LGTM but can you add some documentation as well for the user guide / under labels as this is a new label?

Yes makes sense 👌. Done!

**Comment by cdrage** [comment]
Thank you! Can you show all the options? I think it's cluster, local and blank right?

File: docs/user-guide.md [hunk 1]
```diff
@@ -200,6 +200,7 @@ The currently supported options are:
 | kompose.service.healthcheck.liveness.http_get_path  | kubernetes liveness httpGet path                                                     |
 | kompose.service.healthcheck.liveness.http_get_port  | kubernetes liveness httpGet port                                                     |
 | kompose.service.healthcheck.liveness.tcp_port       | kubernetes liveness tcpSocket port                                                   |
+| kompose.service.external-traffic-policy: local       | kubernetes service external traffic policy                                                   |
```
Comment (cdrage) [inline docs/user-guide.md:203]:
```suggestion
| kompose.service.external-traffic-policy       | 'cluster', 'local', ''                                                   |
```

Let's just do this for now until we update the documentation on user guide. It's a bit of a mess right now to understand / comprehend and I plan to work on this, this month.

**Review by cdrage — CHANGES_REQUESTED** [review]

**Review by cdrage — APPROVED** [review]

**Pull request merged**